spec/models/friendship_spec.rb

```diff
@@ -18,15 +18,15 @@
 	  	end
 
 	  	before(:each) do
-	  		friendship.save
+	  		friendship.save!
 	  	end
 
 	  	it "should increase user's basic_friends by 1 when friendship is created" do
-	  		expect(user.basic_friends).to include(friend)
+	  		expect(user.friends).to include(friend)
```

Comment:
I notice that you aren't using your `factory` here in building `friendship`. I am not sure why the error is happening but I resolved it by doing the ff:

- I changed the way you are building your `friendships` factory:

```ruby
# factories.rb

factory :friendship do
    user
    association :friend, factory: :user
  end
```
- Changed how you built your `friendship` here in the spec:

```ruby
# friendship_spec.rb

let(:friendship) { build(:friendship, user_id: user.id, friend_id: friend.id) }
```

Now running `rspec`, I am not getting any failed tests anymore. When generating all your test data, please make sure you are using factories rather than manually creating them in your spec files. 

I also encourage you to split your factories into separate files:

`/factories/users.rb`
`/factories/friendships.rb`
`/factories/friend_requests.rb`
`/factories/posts.rb`
`/factories/comments.rb`
`/factories/likes.rb`